### ISSUE TITLE
Update Nested Navigation snippet

### ISSUE THREAD

**Issue by huguestennier** [Shopify/slate#237]
### Problem

Currently, Slate default "theme" is using the old `liquid` markup. We would need to update it to support 3 level and also add `js` to support basic keyboard navigation out of the box.

**Comment by NathanPJF** [comment]
Thanks for posting @huguestennier 

What JS for keyboard navigation did you have in mind?  I feel shipping any type of dropdown behaviour in Slate is in scope of the boilerplate because people would need to strip it out if they want different behaviour.

I think nested `<ul>` elements is all that's needed for demo the correct liquid syntax.  Can include `aria-current="page"` as well.

**Comment by huguestennier** [comment]
I am meeting with @t-kelly to discuss further of what we should include in the scope, but what about other aria-attributes à la `aria-controls` ? 

It would be great to provide some keyboard navigation out of the box since it's something usually omitted by developers (and/or have difficulties with).

We could do every interaction with `data-attribute` so you could easily remove them if you want a different behaviour.

Maybe it's out of scope? if so, would there be a way for Shopify to provide developers an easy 3 level deep nav accessible out of the box?

**Comment by jonathanmoore** [comment]
For a Shopify themes baseline requirements, we have to ensure that tabbing properly works to navigate to all levels of the nested nav in sequence.  This also means that shift+tab would reverse the sequence.
![image](http://cloud.jonathanmoore.com/m3u1/Screen%20Recording%202017-08-17%20at%2011.59%20AM.gif)

We utilize `aria-haspopup` to signify that there are nested menus. 
> Indicates that the element has a popup context menu or sub-level menu.
https://www.w3.org/TR/wai-aria/states_and_properties#aria-haspopup

Finally, we set `aria-hidden` to true initially for any nested menus and then using JavaScript set it to false as the tab/focus/hover state changes.
https://www.w3.org/TR/wai-aria/states_and_properties#aria-hidden

**Comment by NathanPJF** [comment]
> It would be great to provide some keyboard navigation out of the box since it's something usually omitted by developers (and/or have difficulties with).

@huguestennier Following up on our IRL convo: I agree this would be useful and is often forgot.  However, I feel shipping a responsive menu with accompanying JS is outside the scope of the boilerplate.  Something to consider in the future, however.

@jonathanmoore thanks for sharing all that 😄  We have the same use of `aria-haspop` in Debut's latest iteration.

**Comment by jonathanmoore** [comment]
Even if the JavaScript is out of scope, the new nested navigation liquid should be implemented.  Is this something that you're waiting on until the new menus go live for everyone?
https://github.com/Shopify/slate/blob/master/packages/slate-theme/src/sections/header.liquid#L81-L104

https://help.shopify.com/themes/development/building-nested-navigation#new-liquid-syntax

**Comment by huguestennier** [comment]
@jonathanmoore Indeed! We plan on updating Slate with the new nested navigation liquid when it's live for everyone, that's why I wanted to discuss the scope of the work before updating it!
